staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/stdlib.go

```diff
@@ -0,0 +1,616 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cel
+
+import (
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/checker/decls"
+	"github.com/google/cel-go/common/types"
+	"github.com/google/cel-go/common/types/ref"
+	"github.com/google/cel-go/common/types/traits"
+	"github.com/google/cel-go/interpreter/functions"
+	exprpb "google.golang.org/genproto/googleapis/api/expr/v1alpha1"
+	"k8s.io/apiextensions-apiserver/third_party/forked/celopenapi/model"
+	"net/url"
+	"regexp"
+)
+
+// Stdlib provides a CEL function library extension that supplements the CEL standard function library.
+//
+// isSorted
+//
+// Returns true if the provided list of comparable elements is sorted, else returns false.
+//
+//     <list<T>>.isSorted() <bool>, T must be a comparable type
+//
+// Examples:
+//
+//     [1, 2, 3].isSorted()  // return true
+//
+//
+// sum
+//
+// Returns the sum of the elements of the provided list. Supports CEL number (int, uint, double) and duration types.
+//
+//     <list<T>>.sum() <T>, T must be a numeric type or a duration
+//
+// Examples:
+//
+//     [1, 3].sum() // returns 4
+//     [].sum() // returns 0
+//
+//
+// min / max
+//
+// Returns the minimum/maximum valued element of the provided list. Supports all comparable types.
+// If the list is empty, an error is returned.
+//
+//     <list<T>>.min() <T>, T must be a comparable type
+//     <list<T>>.max() <T>, T must be a comparable type
+//
+// Examples:
+//
+//     [1, 3].min() // returns 1
+//     [1, 3].max() // returns 3
+//     [].min() // error
+//
+//
+// indexOf / lastIndexOf
```

Comment:
Do we handle float-point precision with the comparison operators?
[1, 2, 3].indexOf(1.00000000000001) == 0

Reply:
Good point. Let me revise the question a bit.
[1.0, 2.0, 3.0].indexOf(1.00000000000001) == 0

Reply:
All of these evaluate to false:

```
[1.0, 2.0, 3.0].indexOf(1.00000000000001) == 0
1.0 == 1.00000000000001
[1.0, 2.0, 3.0].exists(x, x == 1.00000000000001)
```

Signature of indexOf is `<list<T>>.indexOf(<T>) <int>`